src/include/sof/compiler_info.h

```diff
@@ -17,4 +17,14 @@
 #include <arch/compiler_info.h>
 #include <sof/common.h>
 
+/* copy CC_NAME to const arrays during compilation time */
+#define CC_NAME_COPY(field) \
+	field = CC_NAME, \
+	field[ARRAY_SIZE(((struct sof_ipc_cc_version *)(0))->name) - 1] = 0
```

Comment:
Do we really need these macros here? Would like to not clutter `global` space with something so specific like that.
BTW AFAIK literals are null terminated, so why do you do this?

Reply:
@jajanusz they are NULL-terminated, but if you define too long a string, the terminating NULL won't fit so the compiler will happily truncate the string. It might issue a warning though, not sure. As long as we take care to only use string literals, that fit in arrays, we don't need this assignment, but it guards against cases when someone adds too long a string.

Reply:
@lyakh ok, thanks for info, I didn't think about too long strings case